### PR TITLE
Simplify determining if numpy is available

### PR DESCRIPTION
**Pull request by eldruin**
This removes the dependency on setuptools on runtime.

### CODE REVIEW

File: dictdiffer/__init__.py [hunk 1]
```diff
@@ -29,15 +27,11 @@
...
-except pkg_resources.DistributionNotFound:  # pragma: no cover
-    HAS_NUMPY = False
-else:
-    HAS_NUMPY = True
-
     import numpy
-
+    HAS_NUMPY = True
     LIST_TYPES += (numpy.ndarray, )
+except ImportError:  # pragma: no cover
```
Comment (jirikuncar) [inline dictdiffer/__init__.py]: Using `ImportError` is not a good practice. Do you have any issue with using `pkg_resources`?

Comment (eldruin) [reply dictdiffer/__init__.py]: I did not know about `ImportError`being a bad practice. Can you share any links related to this?
We do not ship setuptools so at the moment dictdiffer can only be used in a dev environment on our side.

Comment (jirikuncar) [reply dictdiffer/__init__.py]: As an alternative I would use [`importlib.find_spec`](https://docs.python.org/3/library/importlib.html#importlib.machinery.PathFinder.find_spec) but we would have to remove support for Python 2 which might be still a bit too early for some users.

Comment (jirikuncar) [reply dictdiffer/__init__.py]: > Can you share any links related to this?

It used to be in quantifiedcode.com but the project is dead now. Hence, I don't have anything to support my argument anymore.

**Comment by eldruin** [comment]
Thanks for merging! Do you have any plans for a new release?

**Pull request merged**